errorHandling.java

```diff
@@ -0,0 +1,52 @@
+package sample;
+
+public class ErrorHandling {
+
+  
+  public Object createUser(User user){
+      
+      if(loadUser(user.getId()) instanceof User){
+        // user already exists
+        return USER_ALREADY_EXISTS; // error flag
+        return 101; // or error code
+      };
+      
+      resolveGroup(user);
+      return userRepository.save(user);
+  }
+  
+  public void handleSimulation(Simulation simulation){
+    try{
+      resultHandler.handle(simulation);
+    }
+    catch (Throwable t){
+      log.error("failed to save result", t);
+    }
+  }
+  
```

Comment:
// 3. Provide context
  // Good
  public void validateOwner(String ownerId){
    if(ownerRepository.countTestcasesForOwner(ownerId) == 0){
      throw new ValidationException("owner: " + ownerId + " does not have any testcases. Testcases should be populated...")
    }
  }